src/abi/solana_accounts.rs

```diff
@@ -0,0 +1,430 @@
+                    .from_base58()
+                    .unwrap(),
+            ),
+            SolanaAccount::AssociatedProgramId,
+        ),
+        (
+            BigInt::from_bytes_be(
+                Sign::Plus,
+                &"SysvarRent111111111111111111111111111111111"
+                    .from_base58()
+                    .unwrap(),
+            ),
+            SolanaAccount::Rent,
+        ),
+        (
+            BigInt::from_bytes_be(
+                Sign::Plus,
+                &"TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA"
+                    .from_base58()
+                    .unwrap(),
+            ),
+            SolanaAccount::TokenProgramId,
+        ),
+        (
+            BigInt::from_bytes_be(
+                Sign::Plus,
+                &"SysvarC1ock11111111111111111111111111111111"
+                    .from_base58()
+                    .unwrap(),
+            ),
+            SolanaAccount::ClockAccount,
+        ),
+    ])
+});
+
+impl SolanaAccount {
+    /// Retrieve a name from an account, according to Anchor's constant accounts map
+    /// https://github.com/coral-xyz/anchor/blob/06c42327d4241e5f79c35bc5588ec0a6ad2fedeb/ts/packages/anchor/src/program/accounts-resolver.ts#L54-L60
+    pub(super) fn name(&self) -> String {
+        match self {
+            SolanaAccount::AssociatedProgramId => "associatedTokenProgram".to_string(),
+            SolanaAccount::Rent => "rent".to_string(),
+            SolanaAccount::SystemAccount => "systemProgram".to_string(),
+            SolanaAccount::TokenProgramId => "tokenProgram".to_string(),
+            SolanaAccount::ClockAccount => "clock".to_string(),
+            SolanaAccount::InstructionAccount => "SysvarInstruction".to_string(),
+        }
+    }
+
+    fn from_number(num: &BigInt) -> Option<SolanaAccount> {
+        AVAILABLE_ACCOUNTS.get(num).cloned()
+    }
+}
+
+/// Struct to save the recursion data when traversing all the CFG instructions
+struct RecurseData<'a> {
+    /// Here we collect all accounts. It is a map between the cfg number of a functions and the
+    /// accounts it requires
+    accounts: HashMap<usize, IndexSet<SolanaAccount>>,
+    /// next_queue saves the set of functions we must check in the next iteration
+    next_queue: IndexSet<(usize, usize)>,
+    /// The number of the function we are currently traversing
+    cfg_func_no: usize,
+    /// The contract the function belongs to
+    contract_no: usize,
+    /// The quantity of accounts we have added the the hashmap 'accounts'
+    accounts_added: usize,
+    ns: &'a Namespace,
+}
+
+impl RecurseData<'_> {
+    /// Add an account to the hashmap
+    fn add_account(&mut self, account: SolanaAccount) {
+        let inserted = if let Some(set) = self.accounts.get_mut(&self.cfg_func_no) {
+            set.insert(account)
+        } else {
+            self.accounts
+                .insert(self.cfg_func_no, IndexSet::from([account]));
+            true
+        };
```

Comment:
another nitpick. This could be cleaner with `self.accounts.entry()`, should only a single hash lookup rather than two on insertion.